src/tools/compiletest/src/read2.rs

```diff
@@ -12,28 +12,32 @@ pub struct ReadOutput {
 }
 
 pub fn read2_abbreviated(mut child: Child) -> io::Result<ReadOutput> {
-    use io::Write;
     use std::mem::replace;
 
     const HEAD_LEN: usize = 160 * 1024;
     const TAIL_LEN: usize = 256 * 1024;
 
     enum ProcOutput {
-        Full(Vec<u8>),
+        Full(String),
         Abbreviated { head: Vec<u8>, skipped: usize, tail: Box<[u8]> },
     }
 
     impl ProcOutput {
         fn extend(&mut self, data: &[u8]) {
             let new_self = match *self {
-                ProcOutput::Full(ref mut bytes) => {
-                    bytes.extend_from_slice(data);
-                    let new_len = bytes.len();
+                ProcOutput::Full(ref mut string) => {
+                    let utf8 = String::from_utf8_lossy(data);
```

Comment:
Good catch, pushed a new (simpler) implementation that works over bytes instead of strings, and removed the refactors required to handle strings inside `read2`.